src/main/java/city/parking/controllers/ParkingProcessController.java

```diff
@@ -0,0 +1,74 @@
+package city.parking.controllers;
+
+import city.parking.entities.Money;
+import city.parking.entities.ParkingProcess;
+import city.parking.entities.ParkingProcessPartialUpdateRequest;
+import city.parking.exceptions.ParkingProcessNotFoundException;
+import city.parking.services.ParkingProcessService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@RestController
+@RequestMapping(path = "/parking-processes")
+public class ParkingProcessController {
+    private final ParkingProcessService parkingProcessService;
+
+    public ParkingProcessController(ParkingProcessService parkingProcessService) {
+        this.parkingProcessService = parkingProcessService;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<ParkingProcess>> getParkingProcessesByStage(
+            @RequestParam(name = "stage", required = false) ParkingProcess.Stage processStage) {
+        List<ParkingProcess> processes;
+        if (processStage != null)
+            processes = parkingProcessService.findParkingProcessesByStage(processStage);
+        else {
+            processes = parkingProcessService.findAll();
+        }
+        return ResponseEntity.ok(processes);
+    }
+
+    @GetMapping(value = "/{processId}")
+    public ResponseEntity<ParkingProcess> getParkingProcess(@PathVariable Integer processId) {
+        Optional<ParkingProcess> processOptional = parkingProcessService.findById(processId);
+        if (processOptional.isPresent()) {
+            return ResponseEntity.ok(processOptional.get());
+        } else {
+            throw new ParkingProcessNotFoundException(processId);
+        }
+    }
+
+    @PostMapping
+    public ResponseEntity<ParkingProcess> startParkingProcess(UriComponentsBuilder ucBuilder,
+                                                              @Valid @RequestBody ParkingProcess process) throws URISyntaxException {
+        ParkingProcess newParkingProcess = parkingProcessService.startParkingProcess(process);
+        URI newProcessLocationURI = ucBuilder.path("/parking-processes/{processId}")
+                .buildAndExpand(newParkingProcess.getId()).toUri();
+
+        return ResponseEntity.created(newProcessLocationURI)
+                .body(newParkingProcess);
+    }
+
+    @PatchMapping(value = "/{processId}")
+    public ResponseEntity stopParkingMeter(@PathVariable Integer processId,
```

Comment:
Czemu nie po prostu /{processId}/stop?

Reply:
Ja się nie zgadzam, ale chętnie zobaczymy Twój komentarz:) Można to zrobić też inaczej :)

Reply:
Nad tym elementem zastanawiałam się długo i tak jak opisałam w pliku z założeniami, chciałam, żeby było to 'restful' api, a nie api w z elementem RPC. Dlatego właśnie staram się tu operować na zasobach, wysłać patchem opis docelowego stanu procesu parkowania, zamiast wysyłać po prostu komendę i czasownik w URL'u. 

Z drugiej strony nietrzymanie się kurczowo zasady o operowaniu wyłącznie na zasobach mocno uprościłoby tę operację zatrzymywania parkometru, co pewnie przeważa szalę na korzyść tego rozwiązania